keystone.go

```diff
@@ -180,20 +301,25 @@ func (kClient *KeepaliveKeystoneClient) AddAuthentication(req *http.Request) err
 	}
 
 	if needsRefreshing {
-		kClient.current = nil
+		kClient.tokenID = ""
 	}
 
 	return kClient.KeystoneClient.AddAuthentication(req)
 }
 
 // AddAuthentication adds the authentication data to the HTTP header.
 func (kClient *KeystoneClient) AddAuthentication(req *http.Request) error {
-	if kClient.current == nil {
-		err := kClient.Authenticate()
-		if err != nil {
-			return err
+	if len(kClient.tokenID) == 0 {
```

Comment:
kClient.tokenID == "" is more in golang way (In my opinion)

Reply:
Done